sisl/sparse_geometry.py

```diff
@@ -133,6 +133,10 @@ def copy(self, dtype=None):
         new._csr = self._csr.copy(dtype=dtype)
         return new
 
+    def convert(self, scale):
```

Comment:
add `dtype` here as well, and also.
I think what you want is:
```python
new = self.copy()
new._csr = self._csr.convert(scale, dtype)
return new
```
?
The current implementation returns a `SparseCSR` matrix and not the same class.

Reply:
Hmm, come to think about it. This won't work for spin stuff since the spin value is determined at initialization time...
Hmm...
Perhaps leave this part out so we can play a bit?

Reply:
Leave what out? Not exactly sure what you have in mind.

Reply:
Since it isn't exactly trivial to have the `convert` for the sparse_geometry classes you can just remove everything here. I don't think a user would suspect a Hamiltonian.convert to return a SparseCSR matrix?

Reply:
OK, got it. Indeed the function should not change class.